autoscheduler/user_sessions/views.py

```diff
@@ -97,6 +98,49 @@ def save_availabilities(request):
     """ Saves availabilities for the given user in the session """
     return _set_state_in_session(request, 'availabilities')
 
+@api_view(['GET'])
+def get_saved_schedules(request):
+    """ Returns the saved schedules from the session for the requested term """
+    term = request.query_params.get('term')
+
+    if not term:
+        return Response(status=400)
+
+    with retrieve_data_session(request) as data_session:
+        schedules = data_session.get(term, {}).get('schedules')
+
+        if not schedules:
+            return Response([])
+
+        # Gather all of the section IDs
+        section_ids = set(sec for schedule in schedules for sec in schedule['sections'])
+
+        models = Section.objects.filter(
+            id__in=section_ids
+        ).select_related('instructor').prefetch_related('meetings')
+
+        sections_dict = {section.id: section for section in models.iterator()}
```

Comment:
```suggestion
        sections_dict = {section.id: section for section in models}
```
From [the documentation](https://docs.djangoproject.com/en/3.1/ref/models/querysets/#iterator):
> Also, use of iterator() causes previous prefetch_related() calls to be ignored since these two optimizations do not make sense together.

Also make sure this doesn't perform a query per section